### PR TITLE
Verify JSON schema

### LINKED ISSUE #182: for v1.0, make sure JSON Schema is correct
we need t make sure the JSON Schema (both in the spec and in the test folder) matches the normative descriptions in the spec

### PR DESCRIPTION
**Pull request by rado0x54**
Closes #182. 

- Please note, there were some errors in Schemas and Examples that were corrected.
- Tests were added to all examples
- Ideally it would be nice if tests and spec could draw from the same file(s), currently this is duplicated. 
- Added github actions for tests

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -557,6 +557,8 @@ be ignored:
       ```json
       {
         "presentation_definition": {
+          "id": "32f54163-7166-48f1-93d8-ff217bdb0653",
```
Comment (rado0x54) [inline spec/spec.md:560]: Added required fields.

File: spec/spec.md [hunk 2]
```diff
@@ -200,7 +200,7 @@ requirement.
           }
         ],
         "constraints": {
-          "limit_disclosure": true,
+          "limit_disclosure": "required",
```
Comment (rado0x54) [inline spec/spec.md:203]: Example error.

File: spec/spec.md [hunk 3]
```diff
@@ -669,33 +671,35 @@ values, and an explanation why a certain item or set of data is being requested:
...
-  ],
-  "name": "Verify XYZ Government Employment",
-  "purpose": "We need to know if you currently work at an agency in the XYZ government",
-  "metadata": {
-    "client_id": "40be4fb5-7f3a-470b-aa37-66ed43821bd7",
-    "redirect_uri": "https://tokens.xyz.gov/verify"
-  },
-  "constraints": {
-    "fields": [
+"input_descriptors": [
```
Comment (rado0x54) [inline spec/spec.md:674]: Example was not wrapped in "input_descriptors", like the one next to it.

File: spec/spec.md [hunk 4]
```diff
@@ -1249,6 +1253,8 @@ format-related rules above:
               "enum": ["all", "pick"]
             },
             "count": { "type": "integer", "minimum": 1 },
+            "min": { "type": "integer", "minimum": 0 },
```
Comment (rado0x54) [inline spec/spec.md:1256]: Schema was missing properties. (the overall Presentation Definition Schema had those...)

File: spec/spec.md [hunk 5]
```diff
@@ -1459,7 +1467,7 @@ requisite information to resolve the status of a [[ref:Claim]].
         "filter": {
           "type": "string",
           "format": "date-time",
-          "min": "2020-12-31T23:59:59.000Z"
+          "minimum": "2020-12-31T23:59:59.000Z"
```
Comment (rado0x54) [inline spec/spec.md:1470]: Example mistake

File: spec/spec.md [hunk 6]
```diff
@@ -1918,28 +1927,29 @@ composed and embedded as follows:
...
-            }
+            "format": "jwt_vc",
+            "path": "$.mostInnerClaim[2]"
+          }
         }
-    }
-  ]
+      }
+    ]
+  }
```
Comment (rado0x54) [inline spec/spec.md:1952]: Missing bracket.

**Review by csuwildcat — APPROVED** [review]

**Pull request merged**